frontend/src/components/App/Settings/SettingsCluster.tsx

```diff
@@ -314,6 +314,24 @@ export default function SettingsCluster() {
     );
   }
 
+  // Display the original name of the cluster if it was loaded from a kubeconfig file.
+  function ClusterName() {
+    const originalName = clusterInfo?.meta_data?.originalName;
+    const source = clusterInfo?.meta_data?.source;
+    const isNonDynamic = source === 'kubeconfig';
```

Comment:
Only using the source as a conditional also renders the typography component when there is no originalName. I think it would be better to have something like this:
```suggestion
    // Display original name in the UI when defined
    const displayOriginalName = source === 'kubeconfig' && originalName;
```

Reply:
good idea!

quick question though, were you able to get this to render without an originalName? or is this just a potential issue you mean? 

If you were able to see the typography component render without an original name can you tell me the steps to recreate it, that means the meta_data is either failing or not being completed somewhere and I'd need to know where

Reply:
It was a bug that I noticed but seems like it went away 🤔 unsure how to reproduce

I guess it's also good to make sure there's something to display before rendering it